pytorch_lightning/trainer/connectors/logger_connector.py

```diff
@@ -202,11 +202,11 @@ def _log_on_evaluation_epoch_end_metrics(self, epoch_logs):
             self.log_metrics(metrics_to_log, {}, step=self.trainer.global_step)
 
     def __rename_keys_by_dataloader_idx(self, metrics, dataloader_idx, num_loaders):
-        if num_loaders == 1:
-            return metrics
-
-        result = {f'{k}/dataloader_idx_{dataloader_idx}': v for k, v in metrics.items()}
-        return result
+        if num_loaders != 1:
+            for k in metrics:
+                v = metrics.pop(k)
+                metrics[f'{k}/dataloader_idx_{dataloader_idx}'] = v
+        return metrics
```

Comment:
```
    def __rename_keys_by_dataloader_idx(self, metrics, dataloader_idx, num_loaders):
        if num_loaders == 1:
            return metrics
        # result = Result()
        for k, v in metrics.items():
            list_of_keys = []
            if k == 'meta':
                # result[k] = v
                continue
            else:
                # result[f'{k}/dataloader_idx_{dataloader_idx}'] = v
                list_of_keys.append(k)
        for k in list_of_keys:
            metrics[f'{k}/dataloader_idx_{dataloader_idx}'] = v
            metrics['meta'][f'{k}/dataloader_idx_{dataloader_idx}'] = metrics['meta'][k]
            del(metrics['meta'][k])
        return metrics
```
This is how my method looks like, by renaming the keys to each dataloader but keeping the `meta` key as it is and making sure to remove the previous keys. There's probably cleaner ways, but this worked in my case.